src/shapes/Object/Object.ts

```diff
@@ -2069,35 +2051,19 @@ export const fabricObjectDefaultValues = {
   stateProperties: [
     'top',
     'left',
-    'width',
-    'height',
     'scaleX',
     'scaleY',
     'flipX',
     'flipY',
     'originX',
     'originY',
-    'transformMatrix',
-    'stroke',
-    'strokeWidth',
-    'strokeDashArray',
-    'strokeLineCap',
-    'strokeDashOffset',
-    'strokeLineJoin',
-    'strokeMiterLimit',
     'angle',
     'opacity',
-    'fill',
     'globalCompositeOperation',
     'shadow',
     'visible',
-    'backgroundColor',
     'skewX',
     'skewY',
-    'fillRule',
-    'paintFirst',
-    'clipPath',
-    'strokeUniform',
```

Comment:
removed the overlab between stateful properties and cache properties.
Stateful properties can't be deleted since they are still needed to set group as dirty.